tools/dependency_summary.py

```diff
@@ -0,0 +1,815 @@
+        "name": "openssl",
+        "repository": "https://www.openssl.org/source/",
+        "license": "OpenSSL",
+        "license_file": "https://www.openssl.org/source/license-openssl-ssleay.txt",
+    },
+    "ext-sqlcipher": {
+        "name": "sqlcipher",
+        "repository": "https://github.com/sqlcipher/sqlcipher",
+        "license": "BSD-3-Clause",
+        "license_file": "https://raw.githubusercontent.com/sqlcipher/sqlcipher/master/LICENSE",
+    },
+}
+
+# And these are rust packages that pull in the above dependencies.
+# Others are added on a per-target basis during dependency resolution.
+PACKAGES_WITH_EXTRA_DEPENDENCIES = {
+    "openssl-sys": ["ext-openssl"],
+    "ring": ["ext-openssl"],
+    # As a special case, we know that the "logins" crate is the only thing that enables SQLCipher.
+    # In a future iteration we could check the cargo build-plan output to see whether anything is
+    # enabling the sqlcipher feature, but this will do for now.
+    "logins": ["ext-sqlcipher"],
+}
+
+# Hand-audited tweaks to package metadata, for cases where the data given to us by cargo is insufficient.
+# We list both the expected value from `cargo metadata` and the replacement value, to
+# guard against accidentally overwriting future upstream changes in the metadata.
+#
+# Let's try not to add any more dependencies that require us to edit this list!
+PACKAGE_METADATA_FIXUPS = {
+    # Ring's license describes itself as "ISC-like", and we've reviewed this manually.
+    "ring": {
+        "license": {
+            "check": None,
+            "fixup": "ISC",
+        },
+    },
+    # In this case the rust code is BSD-3-Clause and the wrapped zlib library is under the Zlib license,
+    # which does not require explicit attribution.
+    "adler32": {
+        "license": {
+            "check": "BSD-3-Clause AND Zlib",
+            "fixup": "BSD-3-Clause",
+        },
+        "license_file": {
+            "check": None,
+            "fixup": "LICENSE",
+        }
+    },
+    # These packages do not unambiguously delcare their licensing file.
+    "publicsuffix": {
+        "license": {
+            "check": "MIT/Apache-2.0"
+        },
+        "license_file": {
+            "check": None,
+            "fixup": "LICENSE-APACHE",
+        }
+    },
+    "siphasher": {
+        "license": {
+            "check": "MIT/Apache-2.0"
+        },
+        "license_file": {
+            "check": None,
+            "fixup": "COPYING",
+        }
+    },
+    # These packages do not include their license file in their release distributions,
+    # so we have to fetch it over the network. XXX TODO: upstream bugs to get it included?
+    "argon2rs": {
+        "repository": {
+            "check": "https://github.com/bryant/argon2rs",
+        },
+        "license_file": {
+            "check": None,
+            "fixup": "https://raw.githubusercontent.com/bryant/argon2rs/master/LICENSE",
+        }
+    },
+    "cloudabi": {
```

Comment:
This could also go in the list of 'target specific crates for targets we do not support'